projects/step-core/src/lib/components/restore-dialog/restore-dialog.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { DateTime } from 'luxon';
+import { Observable } from 'rxjs';
+import { History } from '../../client/generated';
+import { TableLocalDataSource } from '../../modules/table/shared/table-local-data-source';
+import { AugmentedResourcesService } from '../../client/augmented/services/augmented-resources-service';
+import { FilterConditionFactoryService } from '../../modules/table/services/filter-condition-factory.service';
+import { SearchColDirective } from '../../modules/table/directives/search-col.directive';
+
+@Component({
+  selector: 'step-restore-dialog',
+  templateUrl: './restore-dialog.component.html',
+  styleUrls: ['./restore-dialog.component.scss'],
+})
+export class RestoreDialogComponent {
+  readonly dataSource: TableLocalDataSource<History>;
+
+  constructor(
+    private _augmentedResourceService: AugmentedResourcesService,
+    private _matDialogRef: MatDialogRef<RestoreDialogComponent>,
+    private _filterConditionFactory: FilterConditionFactoryService,
+    @Inject(MAT_DIALOG_DATA) public plan: { version: string; history: Observable<History[]> }
```

Comment:
I would personally prefer to extract an interface for the data, so I can then pass it when opening the dialog e.g. 
```
const matDialogConfig: MatDialogConfig = {
  data: {...}
};
const dialogRef = this._matDialog.open(DialogComponent, matDialogConfig);
```